### PR TITLE
Added tracking system

### PR DESCRIPTION
**Pull request by ZuluPro**
A cookie actually named `track_key` is used for track visitors.
All are recorded in a `request.tracking.models.Visitor` and  `request.tracking.models.Visit`.
Recorded by `request.middleware.RequestMiddleware` only if `'request.tracking'` is in `INSTALLED_APPS`.

I added a Plugin to show active visitors in overview.
I added features for matching with those definitions: https://en.wikipedia.org/wiki/Web_analytics#On-site_web_analytics_-_definitions

Say me what you think about,
And if it is integrable in django-request or if I should made another repository.
I'll add a lot of plugins for visualize more and more data.

PS: The failing test is `isort`, maybe its configuration isn't enough dynamic ?

### CODE REVIEW

**Comment by felixxm** [comment]
Sort imports correctly. View test [output](https://travis-ci.org/kylef/django-request/jobs/123637520) for more details.

**Comment by ZuluPro** [comment]
Hey @felixxm, I wanted to ping you for ask what do you think about this PR ?

About ``isort``, don't you think it is too strict, it fails for simple things and there's no PEP about, I'vz got few example:
- I wrote: ``from django.test import TestCase, RequestFactory`` before ``from django.http import HttpResponse``
  In my mind, I'm in a test module, so I put ``import TestCase`` before.
- I didn't add blank line between import kinds, but there are only 5 imports, too less I think
- I wrote ``from request.tracking.models import Visitor, Visit`` but it would like ``from request.tracking.models import Visit, Visitor``. Elements sorting is time consuming.

All that for say, I don't think we could say a commit is in error if it doesn't follow the previous examples, particularly the last.

**Comment by ZuluPro** [comment]
@felixxm , @kylef : Any news ?

File: request/middleware.py [hunk 1]
```diff
@@ -30,7 +33,19 @@ def process_response(self, request, response):
             if request.user.username in settings.REQUEST_IGNORE_USERNAME:
                 return response
 
-        r = Request()
-        r.from_http_request(request, response)
+        req = Request()
+        req.from_http_request(request, response)
+
+        if settings.USE_TRACKING:
+            if 'track_key' in request.COOKIES:
```
Comment (kylef) [inline request/middleware.py]: What about making the key customisable? Perhaps there could be a setting for it.

File: request/middleware.py [hunk 2]
```diff
@@ -30,7 +33,19 @@ def process_response(self, request, response):
             if request.user.username in settings.REQUEST_IGNORE_USERNAME:
                 return response
 
-        r = Request()
-        r.from_http_request(request, response)
+        req = Request()
+        req.from_http_request(request, response)
+
+        if settings.USE_TRACKING:
```
Comment (kylef) [inline request/middleware.py]: Can we prefix the setting with `REQUEST_` like the others please.

Comment (kylef) [inline request/middleware.py]: Ah, now I see https://github.com/django-request/django-request/pull/103

File: request/middleware.py (hunk 1 above)
Comment (kylef) [inline request/middleware.py]: I'd also suggest we prefix the key name.

Comment (ZuluPro) [inline request/middleware.py]: Agree with the 2 comments

**Comment by ZuluPro** [comment]
@kylef @felixxm 
I began a fork of django-request for work faster, it is [pony-traffic](https://github.com/ZuluPro/pony-traffic).

If you have ten minutes for test it, it could be great.
This fork has the following features:
- Visit and Visitor as explain in this PR
- GeoIP with a map in the overview
- New plugins for data provided by visit and visitor models
- New traffic plugins (curves in overview chart)
- New admin filters to see specific requests, visitors and visits

These are only options, if you don't add `'request.tracking'` in `INSTALLED_APPS`, you have a classic django-request.

All is not fully accurate, example: the issue referenced here #105.

I really would like to merge that work and I could make it step by step. But simply, what do you think about all that ?

Should we make a separated project ? With a signal in the middleware it will become easier to plugin anything.

**Comment by devdlabs** [comment]
@ZuluPro  I like your idea , what about tracking multiple sites with celery, like sentry does for bug tracking  ?

**Comment by ZuluPro** [comment]
@devdlabs I don't know how sentry do that, but I plan to use Celery for record asynchronously. Thank for your encouragement.

**Pull request closed without merging**